config.proto

```diff
@@ -95,3 +98,9 @@ enum PropagationFormat {
     // see https://www.w3.org/TR/trace-context/
     TRACECONTEXT = 1;
 }
+
+// JavaAgent has the configs specific to javaagent
+message JavaAgent {
+    // opa_evaluator_jar_path is the path to opa evaluator jar
+    google.protobuf.StringValue opa_evaluator_jar_path = 1;
```

Comment:
@davexroth maybe this should be a list?

Reply:
yep, it should be a list.

Also, it should be generic for any jar which implements the filter interface.  Maybe rename to something like `filter_jar_path`.

Reply:
If this is a list we should clarify whether ordering is important, if all of them happen serial and if the result of one affect the others.

Reply:
This list will be used to load jars that implement filter interface, and it does not matter which order they are given. They will all be loaded and checked for blocking.. If any one of the filter implementations returns true to block, the call will be blocked.

Reply:
> If this is a list we should clarify whether ordering is important, if all of them happen serial and if the result of one affect the others.

+1 

the semantics is as follows: block the request if any of the evaluators return true.

Reply:
that said the filters do not mutate state between each other, they should be totally independent.

Reply:
filters can add attributes right?  I think it makes sense to define the order in which the filters run as the order they are defined here.

This would allow one fitter to add an attribute, and subsequent filters to change behavior based on those previously set attributes.

Reply:
> This would allow one fitter to add an attribute, and subsequent filters to change behavior based on those previously set attributes.

The ordering is tricky since we use Java service loader to load filter provider implementations. The service loader does not guarantee the ordering. Though, we could work around it by creating a new classloader per filter jar. 

I am not sure if we want to introduce coupling between filters, the other issue is that there is no read API on the span for attributes.

Reply:
+1 on having decoupled filters. Passing it in a config file sounds very fragile and not explicit. Could you elaborate on the exact use case @davexroth? Has other alternatives (like composing filters) being analyzed?

Reply:
Let's move this forward without explicit ordering. Ordering can be added later once we know why and how. Ordering won't be backwards incompatible.